docs/services/ml/upload-training-script.md

```diff
@@ -0,0 +1,387 @@
+# This parses the required args for the training script.
+# The model_dir variable will contain the output directory where
+# the ML model that this scrips creates should be stored.
+# The data_json variable will contain the metadata for the dataset
+# that you should use to train the model.
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--dataset_file", dest="data_json", type=str)
+    parser.add_argument("--model_output_directory", dest="model_dir", type=str)
+    args = parser.parse_args()
+    model_dir = args.model_dir
+    data_json = args.data_json
+    return model_dir, data_json
+```
+
+{{% /expand%}}
+{{%expand "Step 2: Parse annotations from dataset file" %}}
+
+The `dataset_file` is a file that the Viam platform will pass to the training script when you train an ML model with it.
+The file contains metadata from the dataset used for the training, including the file path for each data point and any annotations associated with the data.
+
+Dataset JSON files for image data sets with bounding box labels and classification labels are formatted as follows:
+
+```json {class="line-numbers linkable-line-numbers"}
+{
+    "image_path": "/path/to/data/data1.jpeg",
+    "bounding_box_annotations":[
+        {
+            "annotation_label":"blue_star",
+            "x_min_normalized":0.38175675675675674,
+            "x_max_normalized":0.5101351351351351,
+            "y_min_normalized":0.35585585585585583,
+            "y_max_normalized":0.527027027027027
+        }],
+    "classification_annotations":[
+        {"annotation_label": "blue_star"}]
+}
+{
+    "image_path": "/path/to/data/data2.jpeg",
+    "bounding_box_annotations":[
+        {
+            "annotation_label":"blue_star",
+            "x_min_normalized":0.2939189189189189,
+            "x_max_normalized":0.4594594594594595,
+            "y_min_normalized":0.25225225225225223,
+            "y_max_normalized":0.5495495495495496
+        }],
+    "classification_annotations": [
+        {
+            "annotation_label": "blue_star"
+        }]
+}
+
+{
+    "image_path": "/path/to/data/data3.jpeg",
+    "bounding_box_annotations":[
+        {
+            "annotation_label":"blue_star",
+            "x_min_normalized":0.03557312252964427,
+            "x_max_normalized":0.2015810276679842,
+            "y_min_normalized":0.30526315789473685,
+            "y_max_normalized":0.5368421052631579
+        },
+        {
+            "annotation_label":"blue_square",
+            "x_min_normalized":0.039525691699604744,
+            "x_max_normalized":0.2015810276679842,
+            "y_min_normalized":0.2578947368421053,"y_max_normalized":0.5473684210526316
+        }],
+    "classification_annotations": [
+        {
+            "annotation_label": "blue_star"
+        },
+        {
+            "annotation_label": "blue_square"
+        }]
+    }
+```
+
+With classification annotations like this:
```

Comment:
I don't know what happened with the comment on this but I was suggesting to remove these? Why do we need them in addition to the above?

Reply:
oh I think I asked Tahiya a question about the first example-- it doesn't make sense to me as to why there's both types of annotations at once in the first example when according to docs "If you want to train an image classifier, use image tags. For an object detector, use bounding boxes." 
is it valid dataset metadata?
she wanted to include the two types distinctly separate and also the example though I think @tahiyasalam

Reply:
It is a valid dataset. We provide a dataset with all possible annotations and the user needs to parse the file and use the annotations that are relevant to them (classifications, bounding boxes, or neither). Since we are not managing the training for them, they need to do this themselves. It's helpful to see the two types, since they're formatted differently and need to be parsed differently (which is why I sent two parse functions originallt).

Reply:
@sguequierre you might use the same dataset for two different training purposes. 

I think we should probably include both parse functions unless we can make one parse function that parses for both and returns an empty array or set for anything it didn't find.

Reply:
I don't feel strongly about keeping them (the json bits) twice! I think originally it was difficult to read, but with the new formatting, it comes across as much clearer.